Content.Shared/Containers/ItemSlot/ItemSlotsSystem.cs

```diff
@@ -255,42 +277,89 @@ private void Insert(EntityUid uid, ItemSlot slot, EntityUid item, EntityUid? use
             _audioSystem.PlayPredicted(slot.InsertSound, uid, excludeUserAudio ? user : null);
         }
 
+        public enum CannotInsertReason
+        {
+            ContainerSlotNull,
+            SlotFullNoSwap,
+            ItemNotWhitelisted,
+            ItemBlacklisted,
+            SlotLocked,
+            EventCanceled,
+
+            /**
+             * calling _containers.CanInsert returned false, i dunno why you figure it out
+             */
+            CanInsertFalse
+        }
+
```

Comment:
Having a big enum hardcoded like this defeats the purpose of the eventbus; I haven't had a chance to read the rest yet but this needs to go.